### DISCUSSION TITLE
Renovate failing to lookup Nuget Packages in private artifactory

### DISCUSSION THREAD

**Discussion by ngishkamau** [renovatebot/renovate#23935] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.14

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

Placeholder value, please select the correct response from the dropdown

### Wanted end result.

We use both public and private hosted packages. In this case, renovate is failing to look up for the nuget dependencies hosted privately

### What you tried so far.

This is the Simple config.js  
 ``````
reviewers: [
  	%gitlab-reviewers%,
  ],
    "hostRules": [
    {
     "hostType": "nuget",
      "matchHost": "https://proget.example.com/feeds/failing-nuget",
      "username":"",
      "password":""
    }

    ],
    onboardingConfigFileName: 'nuget.config',
  repositories: [
    %gitlab-repositories%
  ],
``````

### Relevant debug logs

{
08:46:53 "hostType": "nuget",
08:46:53 "matchHost": "https://progetxxxx.com/feeds/conxxmit-nuget",
08:46:53 "username": "",
08:46:53 "password": ""
08:46:53 }
08:46:53 ],
08:46:53 "onboardingConfigFileName": "nuget.config",
08:46:53 "repositories": [
08:46:53 "pgxxa/foxxa/foxxs/xxns-nuget/xxx-streaming-sdk"
08:46:53 ]
08:46:53 },
08:46:53 "migratedConfig": {
08:46:53 "token": "***********",
08:46:53 "platform": "gitlab",
08:46:53 "detectGlobalManagerConfig": true,
08:46:53 "logFileLevel": "debug",
08:46:53 "onboardingConfig": {"extends": ["config:recommended"]},
08:46:53 "reviewers": ["xx"],
08:46:53 "hostRules": [
08:46:53 {
08:46:53 "hostType": "nuget",
08:46:53 "matchHost": "https://progxxl.com/feeds/coxxt-nuget",
08:46:53 "username": "",
08:46:53 "password": ""
08:46:53 }
08:46:53 ],
08:46:53 "onboardingConfigFileName": "nuget.config",
08:46:53 "repositories": [
08:46:53 "pgfxxa/foxx/forxxs/horizons-nuget/xxmit-streaming-sdk"
08:46:53 ]
08:46:53 }
08:46:54 INFO: Repository started (repository=pgxx/foxxa/fxxlus/horizons-nuget/cxxit-streaming-sdk)
08:46:54 "renovateVersion": "36.51.2"
08:47:01 INFO: Dependency extraction complete (repository=pgxxxa/fxxa/foxxxplus/horizons-nuget/coxxxxit-streaming-sdk, baseBranch=master)
08:47:01 "stats": {
08:47:01 "managers": {
08:47:01 "cake": {"fileCount": 1, "depCount": 1},
08:47:01 "nuget": {"fileCount": 1, "depCount": 14}
08:47:01 },
08:47:01 "total": {"fileCount": 2, "depCount": 15}
08:47:01 }
08:47:04 WARN: Package lookup failures (repository=pgfxxa/forsxx/forxxlus/horizons-nuget/conxxxxt-streaming-sdk)
08:47:04 "warnings": [
08:47:04 "Failed to look up nuget package Coxxit.Cake.Build",
08:47:04 "Failed to look up nuget package Conxxt.Schemas.Sdk",
08:47:04 "Failed to look up nuget package Cxxit.NetCore.Identity.Sdk",
08:47:04 "Failed to look up nuget package Coxxit.Databases",
08:47:04 "Failed to look up nuget package Coxxxit.Configuration"
08:47:04 ],
08:47:04 "files": [
08:47:04 "build.cake",
08:47:04 "code/src/Coxxxt.Streaming.Sdk/Cxxit.Streaming.Sdk.csproj"
08:47:04 ]
08:47:05 INFO: Repository finished (repository=pgxxa/fxx/foxxxs/horizons-nuget/coxxxt-streaming-sdk)
08:47:05 "cloned": true,
08:47:05 "durationMs": 10988

**Comment by rarkins** [comment]
Is your private registry defined in the repository you are running Renovate on? If so then how

**Reply by ngishkamau** [reply]
No,
You I have to define it in the renovate.json?

**Reply by rarkins** [reply]
You will need to define it either in config.js (bot config) or renovate.json (repo config). You can use packageRules with registryUrls - please search this repo for examples

**Comment by ngishkamau** [comment]
``````{
   "$schema":"https://docs.renovatebot.com/renovate-schema.json",
   "extends":[
      "config:base",
      "docker:disable",
      ":dependencyDashboard",
      ":combinePatchMinorReleases",
      ":ignoreUnstable"
   ],
   "timezone":"Europe/Berlin",
   "schedule":[
      "after 5pm every weekday",
      "before 10pm every weekday"
   ],
   "reviewers":[
      
   ],
   "prConcurrentLimit":5,
   "packageRules":[
      {
      "matchDatasources": ["nuget"],
      "registryurl": ["https://prxxxxxxxxt-nuget"]      
    },
         

      {
         "groupName":"all non-major dependencies",
         "matchPackagePatterns":[
            "*"
         ],
         "matchUpdateTypes":[
            "minor",
            "patch"
         ],
         "automerge":false
      },
      {
         "groupName":"all major dependencies",
         "matchPackagePatterns":[
            "*"
         ],
         "matchUpdateTypes":[
            "major"
         ],
         "automerge":false
      },
      {
         "groupName":"Confirmit packages major",
         "matchPackagePatterns":[
            "^Confirmit"
         ],
         "matchUpdateTypes":[
            "major"
         ],
         "automerge":false
      },
      {
         "groupName":"Confirmit packages minor",
         "matchPackagePatterns":[
            "^Confirmit"
         ],
         "matchUpdateTypes":[
            "minor",
            "patch"
         ],
         "automerge":false
      },
      {
         "groupName":"Microsoft",
         "matchPackagePatterns":[
            "^Microsoft"
         ],
         "allowedVersions":"<7",
         "automerge":false
      },
      {
         "groupName":"Refit",
         "matchPackagePatterns":[
            "^Refit"
         ],
         "allowedVersions":"<7",
         "automerge":false
      },
      {
         "groupName":"Serilog",
         "matchPackagePatterns":[
            "^Serilog"
         ],
         "allowedVersions":"<6.0.1",
         "automerge":false
      }
   ],
   "ignoreDeps":[
      "dotnet-ef",
      "JetBrains.Annotations",
      "prometheus-net.AspNetCore",
      "prometheus-net.AspNetCore.HealthChecks"
   ]
}
``````

**Reply by ngishkamau** [reply]
Above is the default.json file with global shareable preset that is extended in every repo.

**Reply by ngishkamau** [reply]
The private endpoint artifactory host is already defined in the config.js. Here is how  do I have to replicate this in the dfault.json file as well

``````
"hostRules": [
    {
     "hostType": "nuget",
      "matchHost": "https://proget2.global.com/feeds/commit-nuget",
      "username":"",
      "password":""
    }
``````

**Reply by rarkins** [reply]
That's hostRules, not packageRules. packageRules with registryUrls are used to tell Renovate where to look for packages. hostRules tell Renovate how to authenticate with a particular host. So you need both.

You have made a typo in the example you copy pasted above, please take more care before trying and commenting

**Reply by ngishkamau** [reply]
Thanks, let me test it out.

**Comment by viceice** [comment]
`onboardingConfigFileName` is wrong, it needs to be one of the supported renovate repo config file names.

**Reply by ngishkamau** [reply]
I have this host rule in the config.js
``````  detectGlobalManagerConfig: true,
  logFileLevel: 'debug',
  onboardingConfig: {
    extends: ['config:base'],
  },
      "hostRules": [
    {
     "hostType": "nuget",
      "matchHost": "https://progxxxxxobal.com/feeds/coxxxxt-nuget",
      "username":"",
      "password":""
    }    
    ],
  reviewers: [
  	%gitlab-reviewers%,
  ],

    onboardingConfigFileName: 'renovate.json',
  repositories: [
    %gitlab-repositories%
  ],
``````


and this matchdatasource rules in the repo json file

`````````json
"packageRules":[

      {
      "matchDatasources": ["nuget"],
      "registryurl":["https://prxxxxxxbal.com/feeds/coxxxit-nuget#protocolVersion=3"],
      "matchPackageNames": ["continamit"]
     
    },
                   
      {
         "groupName":"all non-major dependencies",
         "matchPackagePatterns":[
            "*"

 But still lookup from the private artifactory is still failing is failing. I'm convinced this is the right way to do it, could i be missing something or  the error is caused by other mismatch?

**Reply by rarkins** [reply]
`registyurl` is not a valid option

**Reply by ngishkamau** [reply]
I did replicate this default in my renovate.json file which is global.
````````
{
  "packageRules": [
    {
      "matchDatasources": ["docker"],
      "registryUrls": ["https://docker.mycompany.domain"]
    }
  ]
}
````````

**Comment by ngishkamau** [comment]
Probably I'm missing the contemplation here. I have the host rules in the bot file package rules in a global default.json file which is shareable on organizational level. So every repo has an extending url pointing out to the default.json. should I define this override in the global file or the repo level file.

**Reply by rarkins** [reply, marked as answer]
First step is:
- Is Renovate looking up the right registry URLs? If not then you need packageRules with registryUrls specified
- Is Renovate looking up the right registry URL but failing for authentication? If so then you need hostRules
